src/sentry/sentry_apps/services/hook/impl.py

```diff
@@ -59,3 +63,33 @@ def create_service_hook(
                     hook.add_project(project_id)
 
             return serialize_service_hook(hook)
+
+    def replace_service_hook_projects(
+        self, installation_id: int, project_ids: list[int]
+    ) -> list[RpcServiceHookProject]:
+
+        with transaction.atomic(router.db_for_write(ServiceHookProject)):
+            hook = ServiceHook.objects.get(installation_id=installation_id)
+            ServiceHookProject.objects.filter(service_hook_id=hook.id).delete()
+            hook_projects = []
+            for project_id in project_ids:
+                hook_projects.append(
+                    ServiceHookProject.objects.create(
+                        project_id=project_id,
+                        service_hook_id=hook.id,
+                    )
+                )
+            return [serialize_service_hook_project(hp) for hp in hook_projects]
+
+    def delete_service_hook_projects(self, installation_id: int) -> list[RpcServiceHookProject]:
+
+        hook = ServiceHook.objects.get(installation_id=installation_id)
+        hook_projects = ServiceHookProject.objects.filter(service_hook_id=hook.id)
+        deletions.exec_sync_many(list(hook_projects))
+        return []
```

Comment:
should we be returning the list of deleted `ServiceHookProject` ids, or nothing at all ? I see we returned empty list up top but it doesn't seem too useful here

Reply:
I looked and we seemed to just return 204 in all DELETEs ini existing APIs should I just do the same?

Reply:
got rid of the service and moved code into endpoint (it won't be used by any other endpoints so what's the point - just more complexity, tests and boilerplate)